src/datepicker/ngb-date-struct.ts

```diff
@@ -5,7 +5,14 @@
  *
  * See the [date format overview](#/components/datepicker/overview#date-model) for more details.
  */
-export interface NgbDateStruct {
+export interface NgbDateStruct extends NgbMonthStruct {
+  /**
+   * The day of month, starting at 1
+   */
+  day: number;
+}
+
+export interface NgbMonthStruct {
```

Comment:
Let's remove this new structure. People are already not very happy with `NgbDateStruct` :)

Let's use `month: {month: number, year: number, day?: number}` for public APIs, it will consume everything